packages/tools/client-debugger/client-debugger/src/messaging/AudienceMessages.ts

```diff
@@ -0,0 +1,42 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { IClient } from "@fluidframework/protocol-definitions";
+import { AudienceChangeLogEntry } from "../Logs";
+import { HasContainerId } from "./DebuggerMessages";
+
+import { IDebuggerMessage } from "./Messages";
+
+/**
+ * Inbound event requesting {@link AudienceChangeLogEntry} of the Container with the specific ID.
+ * Will result in the {@link AudienceEventMessage} message being posted.
+ *
+ * @public
+ */
+export interface GetAudienceMessage extends IDebuggerMessage<HasContainerId> {
+	type: "GET_AUDIENCE_EVENT";
+}
+
+/**
+ * Message data format used by {@link AudienceEventMessage}.
+ * @public
+ */
+export interface AudienceEventMessageData extends HasContainerId {
+	/**
+	 * Contents of the Audience event
+	 */
+	allAudienceClientId: string[];
```

Comment:
I don't think we need this. These IDs are available via each `IClient`'s `user` property via `client.user.id`.

What I think we want instead is the client ID of **this** session's user, which can be accessed via `container.clientId`. Note that this data changes between connects and disconnects.

Reply:
`allAudienceClientId` is removed and `AudienceEventMessageData` now contains:
- `clientId: string | undefined;`
- `audienceState: AudienceClientMetaData[];`
- `audienceHistory: readonly AudienceChangeLogEntry[];`